src/software/gui/full_system/widgets/ai_control.h

```diff
@@ -11,37 +11,103 @@
+    std::shared_ptr<Parameter<bool>> friendly_colour_yellow_parameter);
 
 /**
  * Sets up the ComboBox widget that will be used to select the side the friendly team
  * is defending
+ *
+ * @param widget The widget containing the ComboBox to set up
+ * @param defending_side_override_parameter The parameter that controls whether or not
+ * we override the defending side
+ * @param defending_positive_side_parameter The parameter that determines if the we
+ * are defending the positive side of the field or not
  */
-void setupDefendingSideComboBox(Ui::AutogeneratedFullSystemMainWidget *widget,
-                                std::shared_ptr<ThunderbotsConfig> config);
+void setupDefendingSideComboBox(
+    Ui::AutogeneratedFullSystemMainWidget *widget,
+    std::shared_ptr<Parameter<bool>> defending_side_override_parameter,
+    std::shared_ptr<Parameter<bool>> defending_positive_side_parameter);
 
 /**
  * Sets up the ComboBox widget that will be used to override the AI GameState
+ *
+ * @param widget The widget containing the ComboBox to set up
+ * @param gamestate_override_parameter The parameter that controls whether or not
+ * we override the gamestate
+ * @param previous_gamestate_parameter The parameter that holds the previous gamestate
+ * @param current_gamestate_parameter The parameter that holds the current gamestate
  */
-void setupGameStateOverrideComboBox(Ui::AutogeneratedFullSystemMainWidget *widget,
-                                    std::shared_ptr<ThunderbotsConfig> config);
+void setupGameStateOverrideComboBox(
+    Ui::AutogeneratedFullSystemMainWidget *widget,
+    std::shared_ptr<Parameter<bool>> gamestate_override_parameter,
+    std::shared_ptr<Parameter<std::string>> previous_gamestate_parameter,
+    std::shared_ptr<Parameter<std::string>> current_gamestate_parameter);
 
 /**
  * Sets up the ComboBox widget that will be used to override the current AI Play
+ *
+ * @param widget The widget containing the ComboBox to set up
+ * @param play_override_parameter The parameter that controls whether or not
+ * we override the AI Play
+ * @param current_play_parameter The parameter that holds the current AI Play
  */
-void setupPlayOverrideComboBox(Ui::AutogeneratedFullSystemMainWidget *widget,
-                               std::shared_ptr<ThunderbotsConfig> config);
+void setupPlayOverrideComboBox(
+    Ui::AutogeneratedFullSystemMainWidget *widget,
+    std::shared_ptr<Parameter<bool>> play_override_parameter,
+    std::shared_ptr<Parameter<std::string>> current_play_parameter);
+
+/**
+ * Sets up the ComboBox widget that will be used to set the communication channel
+ *
+ * @param widget The widget containing the SpinBox to set up
+ * @param channel_parameter The parameter that controls the communication channel
+ */
+void setupChannelSpinBox(Ui::AutogeneratedFullSystemMainWidget *widget,
+                         std::shared_ptr<Parameter<int>> channel_parameter);
```

Comment:
Not blocking since this is probably out of scope, but should the channel be an `unsigned int`?

Reply:
Yes, but those aren't currently supported by Dynamic Params. The min value of the spinbox range is set to `0` for now which will prevent people from setting negative values in the GUI.